invenio_communities/members/services/service.py

```diff
@@ -290,21 +299,16 @@ def _invite_factory(self, identity, community, role, visible, member, message, u
                 uow=uow,
             )
 
-            # add role as message
-            data = {
-                "payload": {
-                    "content": _('You will join as "{role}".').format(role=role.title),
-                }
-            }
-            current_events_service.create(
-                identity, request_item.id, data, CommentEventType, uow=uow
-            )
-            # message was provided.
-            if message:
-                data = {"payload": {"content": message}}
-                current_events_service.create(
-                    identity, request_item.id, data, CommentEventType, uow=uow
+            uow.register(
+                NotificationOp(
+                    CommunityInvitationSubmittedNotificationBuilder.build(
+                        request=request_item._request,
+                        # explicit string conversion to get the value of LazyText
+                        role=str(role.title),
```

Comment:
It is not part of the request, but it will get set on the member entry. Could probably fetch it from there but requires another service method call.


Now that the description will be `You will join as <role>`, we could also use this in the template and get rid of the role. Something like:

```
Invitation to join community '{{ community }}'. {{ request.description }}
```

What do you think?

Reply:
Generally: Which language will be used for the translation of `You will join as <role>`? Languages set by the user creating the invitation and the one receiving the invitation might differ.

Reply:
from where do we get the language set by the user? I don't find it fetched in here, is it in another place?